deep_qa/data/instances/tuple_inference_instance.py

```diff
@@ -0,0 +1,443 @@
+                 # (num_answer_options, num_answer_tuples, num_tuple_slots, num_slot_words)
+                 answers_indexed: List,
+                 # (num_background_tuples, num_tuple_slots, num_slot_words)
+                 background_indexed: List,
+                 label,
+                 index: int=None):
+        super(IndexedTupleInferenceInstance, self).__init__(label, index)
+        self.answers_indexed = answers_indexed
+        self.background_indexed = background_indexed
+
+    @classmethod
+    @overrides
+    def empty_instance(cls):
+        return IndexedTupleInferenceInstance([], [], label=None, index=None)
+
+    @overrides
+    def get_lengths(self) -> Dict[str, int]:
+        # We care only about the longest slot here because all slots will be padded to the same length.
+        max_answer_length = 0
+        max_word_length = 0
+        max_num_slots = 0
+        max_question_tuples = 0
+        for answer in self.answers_indexed:
+            answer_slots, answer_length, answer_word_length = self.get_lengths_from_indexed_tuples(answer)
+            max_num_slots = max(max_num_slots, answer_slots)
+            max_answer_length = max(max_answer_length, answer_length)
+            max_word_length = max(max_word_length, answer_word_length)
+            max_question_tuples = max(max_question_tuples, len(answer))
+        background_slots, background_length, background_word_length = \
+            self.get_lengths_from_indexed_tuples(self.background_indexed)
+        max_num_slots = max(max_num_slots, background_slots)
+        max_word_length = max(max_word_length, background_word_length)
+
+        lengths = {'num_options': len(self.answers_indexed),
+                   'num_question_tuples': max_question_tuples,
+                   'num_background_tuples': len(self.background_indexed),
+                   'question_slot_length': max_answer_length,
+                   'background_slot_length': background_length,
+                   'num_slots': max_num_slots}
+        if max_word_length > 0:
+            lengths['word_character_length'] = max_word_length
+
+    @staticmethod
+    def get_lengths_from_indexed_tuples(indexed_tuples: List):
+        '''
+        Helper method for self.get_lengths().  This gets the max lengths (max number of slots and max
+        number of words per slot) for a list of indexed tuples.
+
+        Parameters
+        ----------
+        indexed_tuples: List[List[List[int]]]
+            The indexed tuples that we want to get max lengths from.
+
+        Returns
+        -------
+        max_num_slots: int
+            The max number of slots found.
+
+        max_slot_words: int
+            The max number of words found in any of the slots.
+
+        max_word_length: int
+            The length of the longest word, with a return value of -1 if the model isn't using character padding.
+        '''
+        max_num_slots = 0
+        max_slot_words = 0
+        max_word_length = 0
```

Comment:
Ok, I see, you check for it above, so it's probably ok.  The default value doesn't match the docstring, though.